addons/web/static/src/js/view_form.js

```diff
@@ -6310,6 +6310,44 @@ instance.web.form.StatInfo = instance.web.form.AbstractField.extend({
 
 });
 
+/**
+    This widget is intended to be used on boolean fields. It toggles a button
+    siwtching between a green bullet / gray bullet.
+*/
+instance.web.form.FieldToggleBoolean = instance.web.form.AbstractField.extend({
+    init: function (field_manager, node) {
+        this._super.apply(this, arguments);
+        this.icon = "";
+    },
+    render_value: function () {
+        var self = this;
+        self.icon = self.get_value() ? 'gtk-yes' : 'gtk-normal';
+        self.$el.html(QWeb.render("toggle_button", {'widget': self, 'prefix': instance.session.prefix,}));
+        self.$el.find('button').on('click', self.set_toggle_button.bind(self));
+    },
```

Comment:
Well, I agree, however this creates some issues with the widget not being always correctly rendered. I suspect some issues in the web client about correctly triggering and rendering widgets in view mode. I keep the render_value while no better (notably involving less code) solution is found.